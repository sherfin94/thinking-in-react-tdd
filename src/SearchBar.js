import React, {Component} from 'react'

export default class SearchBar extends Component {
  handleChange() {
    this.props.handleChange(this.refs.filterTextInput.value);
  }
  render() {
    return(
      <form>
        <input type = "textbox" ref='filterTextInput' onChange={this.handleChange.bind(this)} value={this.props.filterText} />
      </form>
    )
  }
}
