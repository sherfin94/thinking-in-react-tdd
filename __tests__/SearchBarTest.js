import React from 'react';
import SearchBar from '../src/SearchBar';
import renderer from 'react-test-renderer';
//import { mount } from 'enzyme';

describe("Search Bar", () => {
  it("snapshot", () => {
    const component = renderer.create(
      <SearchBar />
    );
    let json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it(" handles onChange event", function() {
    
  })
})
