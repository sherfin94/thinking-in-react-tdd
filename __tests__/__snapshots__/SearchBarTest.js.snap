exports[`Search Bar snapshot 1`] = `
<form>
  <input
    onChange={[Function bound handleChange]}
    type="textbox"
    value={undefined} />
</form>
`;
